src/make_ntuples.c

```diff
@@ -62,14 +62,26 @@ double get_tof(REC_Scintillator rsci, REC_Calorimeter  rcal, int pindex) {
     return tof;
 }
 
-int run(char * in_filename, bool debug, int nevn, int run_no, double beam_E) {
-    // Extract sampling fraction parameters.
-    double sf_params[NSECTORS][SF_NPARAMS][2];
-    if (get_sf_params(Form("../data/sf_params_%06d.root", run_no), sf_params)) return 8;
+int run(char * in_filename, bool use_simul,bool debug, int nevn, int run_no, double beam_E) {
+    double sf_params[NSECTORS][SF_NPARAMS][2] = {{{0.246,0.0140},{0.985,1.},{0.083,0.},{-0.031,0.}},
+                                                 {{0.255,0.0150},{0.985,1.},{0.043,0.},{-0.024,0.}},
+                                                 {{0.245,0.0130},{0.985,1.},{0.067,0.},{-0.025,0.}},
+                                                 {{0.245,0.0140},{0.985,1.},{0.048,0.},{-0.016,0.}},
+                                                 {{0.252,0.0130},{0.985,1.},{0.007,0.},{-0.007,0.}},
+                                                 {{0.242,0.0140},{0.985,1.},{0.095,0.},{-0.038,0.}}};
+    char*  out_filename = (char *) malloc(128 * sizeof(char));
+    if(!use_simul){
+        // Extract sampling fraction parameters.
+        if (get_sf_params(Form("../data/sf_params_%06d.root", run_no), sf_params)) return 8;
```

Comment:
The run number error I mentioned before happens in this file too, causing the program to end its execution early at this point. I cannot execute the program due to this error, so please fix all review comments up to this point so that I can continue the review.

Reply:
Fixed!

Reply:
Great, thanks! I'll be out of town during the weekend but I'll get back to checking the PR on Monday.